cleanenv.go

```diff
@@ -459,15 +460,27 @@ func readEnvVars(cfg interface{}, update bool) error {
 func parseValue(field reflect.Value, value, sep string, layout *string) error {
 	// TODO: simplify recursion
 
+	valueType := field.Type()
+
+	// look for supported struct parser
+	if structParser, found := validStructs[valueType]; found {
+		return structParser(&field, value, layout)
+	}
+
```

Comment:
What is the meaning of moving this code block here?

Reply:
Standard type struct has `encoding.TextUnmarshaler` implementation.
For example `time.Time`: https://cs.opensource.google/go/go/+/master:src/time/time.go;l=1385
Therefore parsing the structure is higher than the implementation check `encoding.TextUnmarshaler`.

Reply:
Okay, that makes sense.

Reply:
I would appreciate a comment describing that, so someone will not accidentally brake it in the future by moving this block down.

Reply:
Ok, done ✅